doc/tutorials/enable_software_sram.rst

```diff
@@ -0,0 +1,129 @@
+
+`CRL` The cache reservation library (CRL) contains the logic required to create software SRAM regions within the physical address space.
+
+`ABI` Application Binary Interface, the CRL binary is delivered via the BIOS and has a well-defined Application Binary Interface (ABI).
+
+`RTCD` Real-time configuration data (RTCD) contains settings for how the system should be optimized for improved temporal performance.
+
+`RTCT` The real-time configuration table (RTCT) enables software to programmatically discover which Intel® TCC resources are available and how they “should” be configured (as indicated by the data in RTCD). For the software SRAM capability, RTCT provides the instructions to the CRL about size and location of software SRAM regions to establish.
+
+Software SRAM Work Flow
+=======================
+
+1. TCC offline tool configures Software SRAM resources, indicated by the data in RTCD and RTCT.
+
+2. ACRN hypervisor parses RTCT and initializes software SRAM by calling CRL ABI on each physical CPU.
+
+3. CRL initializes software SRAM by setting up cache ways and mapping locked cache to memory space.
+
+4. ACRN hypervisor:
+	- Reserve the address range for software SRAM and remap software SRAM region to guest address space.
+	- Build and inject virtual RTCT(vRTCT) to RTVM as an ACPI table.
+	- Setup protection for software SRAM from being flushed runtimely, such as trap and emulate `wbinvd` instruction from guest.
+
+5. Real-Time Configuration Driver (RTVM Kernel) initializes when valid RTCT is detected and creates device node `/dev/tcc_buffer`.
+
+6. User space applications or libraries query configurations of software SRAM and request software SRAM buffers by issuing command to `/dev/tcc_buffer`.
+
+7. User space applications or libraries map software SRAM bufer to user space by issuing command to `/dev/tcc_buffer`.
+
+8. User space applications can access the low-latency software SRAM buffers.
+
+Design assumptions
+==================
+
+1. One RTVM at most, either pre-launch RTVM or pre-launch RTVM, can be supported for specific ACRN configuration. Will update this design when ACRN supports multiple RTVMs.
+
+2. Software SRAM and CAT features can't co-exist for current design, only one of them is allowed for one ACRN configuration.
+
+3. So far, software SRAM can be supported on Intel® TGL and EHL platforms only.
+
+4. When software SRAM is enabled for pre-launch RTVM, ACRN pass-through all software SRAM resource to this RTVM， which means:
+   - Platform RTCT instead of vRTCT will be injected to RTVM.
+   - RTVM access software SRAM region by host physical address(HPA), or say, identical mapping is used for case.
+
+5. When trying to enable software SRAM for pre-launch RTVM, L2 cache software SRAM is NOT supported， or say, ONLY L3 cache can be configured in platform RTCT.
```

Comment:
What do you mean by "trying to...", it sounds like it may fail?

Where is the L3 cache only limitation coming from? HW or ACRN SW implementation?

Reply:
will add one more note on this, thanks.